Create separate shader pipeline for each use case (#311)
membraneframework/video_compositor: docs/pages/concept/shaders.md

The rectangle formed by these triangles spans the whole clip space, i.e. [-1, 1] X [-1, 1].

Each video passed in as input gets a separate rectangle.
`plane_id` in `base_params` `push_constant` represents number of currently rendered plane (texture).

> If there are no input textures, `plane_id` is equal to -1 and a single rectangle is passed into shader. It's useful for shaders only generating something in fragment shader.

Reviewer: ```suggestion
If there are no input textures, `plane_id` is equal to -1 and a single rectangle is passed to the shader. It is only useful for shaders that generate something in the fragment shader.
```
